containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -77,71 +109,163 @@ def get_account_id() -> str:
-        """
-        if bucket_name is None:
-            bucket_name = get_bucket_name()
+    name: str
+    data_product: DataProductConfig
 
-        self.name = name
+    @staticmethod
+    def load(name, data_product_name):
+        data_product = DataProductConfig(name=data_product_name)
+        return DataProductElement(data_product=data_product, name=name)
 
-        self.raw_data_prefix = BucketPath(
-            bucket=bucket_name, key=os.path.join("raw_data", name, table_name) + "/"
+    @property
+    def raw_data_prefix(self):
+        """
+        The path to the raw data in s3 up to and including the element name,
+        e.g. raw_data/my-data-product/some-element/
+        """
+        return BucketPath(
+            bucket=self.data_product.raw_data_bucket,
+            key=os.path.join("raw_data", self.data_product.name, self.name) + "/",
         )
 
-        self.curated_data_prefix = BucketPath(
-            bucket_name,
+    @property
+    def curated_data_prefix(self):
+        """
+        The path to the curated data in s3 up to and including the element name,
+        e.g. curated_data/database_name=my-data-product/table_name=some-element/
+        """
+        return BucketPath(
+            self.data_product.curated_data_bucket,
             os.path.join(
                 "curated_data",
-                f"database_name={name}",
-                f"table_name={table_name}",
+                f"database_name={self.data_product.name}",
+                f"table_name={self.name}",
             )
             + "/",
         )
 
-        self.raw_data_table = QueryTable(
-            database=RAW_DATABASE_NAME, name=f"{table_name}_raw"
-        )
-        self.curated_data_table = QueryTable(database=name, name=table_name)
+    def raw_data_table_unique(self):
+        """
+        A unique table name to use for querying raw data via athena.
+        These tables are always temporary.
+        E.g. ('data_products_raw', 'some_element_3d95ff89-b063-484d-b510-53742d0a6a64_raw)
+        """
+        suffix = uuid4()
+        name = f"{self.name}_{suffix}_raw"
+        if len(name) > MAX_IDENTIFIER_LENGTH:
+            raise ValueError(f"Generated table name too long: {name}")
+
+        return QueryTable(database=RAW_DATABASE_NAME, name=name)
+
+    @property
+    def curated_data_table(self):
+        """
+        The name of the athena table for the data product element.
+        E.g. ('my_data_product', 'some_element')
+        """
+        return QueryTable(database=self.data_product.name, name=self.name)
 
     def raw_data_path(self, timestamp: datetime, uuid_value: UUID) -> BucketPath:
         """
         Path to the raw data extracted at a particular timestamp.
+        E.g. raw_data/my-data-product/some-element/extraction_timestamp=20230101T000000Z/3d95ff89-b063-484d-b510-53742d0a6a64
         """
         return self.extraction_config(timestamp, uuid_value).path
```

Comment:
```suggestion
        return self.extraction_instance(timestamp, uuid_value).path
```
If changing [method name](https://github.com/ministryofjustice/data-platform/pull/1549/files#diff-f47b07dfcbf000013d2c9d5ec4009c2b492803a64344d8da662490171a6184dcR182), needs changing here too

Reply:
Thanks, good spot